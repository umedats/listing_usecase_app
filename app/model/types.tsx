import { Subscription as SDKSubscription } from "@/app/sdk/notifications";

/**
 * The Issue returned from the Listing API.
 */
export interface Issue {
  code: string;
  message: string;
  severity: string;
  attributeNames?: string[];
}

/**
 * The result of the Listings Items API submission.
 */
export interface ListingSubmissionResult {
  sku: string;
  status: string;
  submissionId: string;
  issues?: Issue[];
}

/**
 * Relevant information from Listings GET API call.
 */
export interface Listing {
  attributes: object;
  issues?: Issue[];
  productType?: string;
}

/**
 * A type which represents a patch for the contribution.
 */
export interface Patch {
  op: string;
  path: string;
  value: any;
}

/**
 * A type for the alert dialog state.
 */
export interface AlertDialogState {
  showAlert: boolean;
  alertTitle: string;
  alertContent: string;
}

/**
 * A type which captures the SP API request and response.
 */
export interface SPAPIRequestResponse {
  apiName: string;
  apiDocumentationLink: string;
  request: object;
  response: object;
}

/**
 * A type which capture the information about a feed.
 */
export interface Feed {
  feedId: string;
  feedType: string;
  createdTime: string;
  processingStatus: string;
  resultFeedDocumentId: string;
}

/**
 * A type which capture the information about a Subscription.
 */
export interface Subscription {
  notificationType: string;
  eventBusName?: string;
  eventBusFilterRule?: string;
  sqsTargetId?: string;
  sqsQueueName?: string;
  subscriptionId: string;
}

/**
 * A type which captures the raw SDK subscription along with the
 * notification type.
 */
export interface RawSubscription {
  notificationType: string;
  subscription: SDKSubscription;
}
