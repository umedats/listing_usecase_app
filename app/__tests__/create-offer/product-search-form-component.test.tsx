import "@testing-library/jest-dom";
import {
  render,
  fireEvent,
  act,
  screen,
  cleanup,
} from "@testing-library/react";
import React from "react";
import { NextIntlProvider } from "next-intl";
import ProductSearchFormComponent from "@/app/[locale]/create-offer/product-search-form-component";
import { afterEach } from "@jest/globals";
import { SWRConfig } from "swr";
import userEvent from "@testing-library/user-event";
import { mockResolveFetchResponse } from "@/app/test-utils/mock-fetch";
import { MOCK_CATALOG_ITEMS_API_RESPONSE } from "@/app/test-utils/mock-catalog-items-result";
import fetch from "jest-fetch-mock";
import DebuggingContextProvider, {
  DebugState,
} from "@/app/context/debug-context-provider";
import { SPAPIRequestResponse } from "@/app/model/types";
import { MOCK_SP_API_RESPONSE } from "@/app/test-utils/mock-debug-context";
import translations from "@/app/internationalization/translations/en-US.json";
import { US_LOCALE } from "@/app/constants/global";

beforeEach(() => {
  fetch.resetMocks();
});

// Define your global afterEach function
afterEach(() => {
  // Cleanup any resources or state after each test
  cleanup();
});

function renderProductSearchFormComponent(
  handleSearchRequestSubmit: jest.Mock<any, any, any>,
  debugState?: DebugState,
) {
  return render(
    <SWRConfig value={{ provider: () => new Map() }}>
      <DebuggingContextProvider initialDebugState={debugState}>
        <NextIntlProvider locale={US_LOCALE} messages={translations}>
          <ProductSearchFormComponent
            handleSearchRequest={handleSearchRequestSubmit}
          />
        </NextIntlProvider>
      </DebuggingContextProvider>
    </SWRConfig>,
  );
}

describe("Test for the ProductSearchFormComponent", () => {
  test("Snapshot Test for product search form with empty input", () => {
    // Define a mock handleSearchRequest function
    const handleSearchRequestSubmit = jest.fn();
    const { asFragment } = renderProductSearchFormComponent(
      handleSearchRequestSubmit,
    );
    expect(asFragment()).toMatchSnapshot();
  });

  test("Snapshot Test for product search form with both identifier and keyword input invalid page", () => {
    // Define a mock handleSearchRequestSubmit function
    const handleSearchRequestSubmit = jest.fn();

    const { asFragment, getByLabelText } = renderProductSearchFormComponent(
      handleSearchRequestSubmit,
    );

    // Modify the input fields
    const identifiersInput = getByLabelText("Identifiers");
    fireEvent.change(identifiersInput, { target: { value: "UPC_testtest" } });

    const keywordsInput = getByLabelText("Keywords");
    fireEvent.change(keywordsInput, { target: { value: "keywords_testtest" } });

    expect(asFragment()).toMatchSnapshot();
  });

  test("Snapshot Test for product search form with valid input", () => {
    // Define a mock handleSearchRequestSubmit function
    const handleSearchRequestSubmit = jest.fn();

    const { asFragment, getByLabelText } = renderProductSearchFormComponent(
      handleSearchRequestSubmit,
    );

    // Modify the input fields
    const identifiersInput = getByLabelText("Identifiers");
    fireEvent.change(identifiersInput, { target: { value: "UPC_testtest" } });

    // Trigger the form submission
    act(() => {
      fireEvent.submit(screen.getByTestId("productSearchRequestSubmit")); // Find form by test ID attribute
    });

    expect(asFragment()).toMatchSnapshot();
  });

  test("Snapshot Test for product search form with both identifier and keyword input combinations", async () => {
    // Define a mock handleSearchRequestSubmit function
    const handleSearchRequestSubmit = jest.fn();
    const routeContext = new Map<string, SPAPIRequestResponse[]>();

    mockResolveFetchResponse(200, {
      data: MOCK_CATALOG_ITEMS_API_RESPONSE,
      debugContext: [MOCK_SP_API_RESPONSE],
    });

    const { asFragment, getByLabelText } = renderProductSearchFormComponent(
      handleSearchRequestSubmit,
      {
        routeContext: routeContext,
      },
    );

    // Modify the input fields
    const identifiersInput = getByLabelText("Identifiers");
    await userEvent.type(identifiersInput, "UPC_testtest");
    await userEvent.clear(identifiersInput);

    const keywordsInput = getByLabelText("Keywords");
    await userEvent.type(keywordsInput, "keywords_testtest");
    await userEvent.clear(keywordsInput);

    await userEvent.type(keywordsInput, "keywords_testtest");
    await userEvent.type(identifiersInput, "UPC_testtest");

    await userEvent.clear(keywordsInput);

    expect(routeContext.size).toStrictEqual(0);
    await act(async () => {
      await userEvent.click(screen.getByTestId("submitButton"));
    });
    expect(routeContext.size).toStrictEqual(1);
    expect(asFragment()).toMatchSnapshot();
  });
});
