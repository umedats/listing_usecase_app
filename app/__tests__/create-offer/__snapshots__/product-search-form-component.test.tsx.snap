// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test for the ProductSearchFormComponent Snapshot Test for product search form with both identifier and keyword input combinations 1`] = `
<DocumentFragment>
  <form
    data-testid="productSearchRequestSubmit"
  >
    <div
      class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-3 css-18ksmt-MuiGrid-root"
    >
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-2 css-63b4os-MuiGrid-root"
      >
        <label
          aria-label="Comma-delimited list of product identifiers to search the Amazon catalog for"
          class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-animated MuiFormLabel-colorPrimary MuiInputLabel-root MuiInputLabel-animated css-9npbnl-MuiFormLabel-root-MuiInputLabel-root"
          data-mui-internal-clone-element="true"
          for="identifiersInput"
        >
          Identifiers
        </label>
      </div>
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6 MuiGrid-grid-sm-4 css-1xd74lj-MuiGrid-root"
      >
        <div
          class="MuiFormControl-root MuiFormControl-marginNormal MuiTextField-root css-1rv3ei8-MuiFormControl-root-MuiTextField-root"
        >
          <div
            class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-formControl MuiInputBase-sizeSmall css-9ddj71-MuiInputBase-root-MuiOutlinedInput-root"
          >
            <input
              aria-invalid="false"
              class="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputSizeSmall css-1n4twyu-MuiInputBase-input-MuiOutlinedInput-input"
              id="identifiersInput"
              type="text"
              value="UPC_testtest"
            />
            <fieldset
              aria-hidden="true"
              class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
            >
              <legend
                class="css-ihdtdm"
              >
                <span
                  class="notranslate"
                >
                  ​
                </span>
              </legend>
            </fieldset>
          </div>
        </div>
      </div>
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6 MuiGrid-grid-sm-6 css-1osj8n2-MuiGrid-root"
      >
        <div
          class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-colorPrimary MuiInputBase-sizeSmall  css-74e2me-MuiInputBase-root-MuiInput-root-MuiSelect-root"
        >
          <div
            aria-expanded="false"
            aria-haspopup="listbox"
            aria-labelledby="identifiersType"
            class="MuiSelect-select MuiSelect-standard MuiInputBase-input MuiInput-input MuiInputBase-inputSizeSmall css-jd1zyo-MuiSelect-select-MuiInputBase-input-MuiInput-input"
            id="identifiersType"
            role="button"
            tabindex="0"
          >
            UPC
          </div>
          <input
            aria-hidden="true"
            aria-invalid="false"
            class="MuiSelect-nativeInput css-yf8vq0-MuiSelect-nativeInput"
            tabindex="-1"
            value="UPC"
          />
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiSelect-icon MuiSelect-iconStandard css-pqjvzy-MuiSvgIcon-root-MuiSelect-icon"
            data-testid="ArrowDropDownIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M7 10l5 5 5-5z"
            />
          </svg>
        </div>
      </div>
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-2 css-63b4os-MuiGrid-root"
      >
        <label
          aria-label="Comma-delimited list of words or product identifiers to search the Amazon catalog for."
          class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-animated MuiFormLabel-colorPrimary MuiInputLabel-root MuiInputLabel-animated css-9npbnl-MuiFormLabel-root-MuiInputLabel-root"
          data-mui-internal-clone-element="true"
          for="keywordsInput"
        >
          Keywords
        </label>
      </div>
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-6 css-1b3l6lk-MuiGrid-root"
      >
        <div
          class="MuiFormControl-root MuiFormControl-marginNormal MuiTextField-root css-1rv3ei8-MuiFormControl-root-MuiTextField-root"
        >
          <div
            class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-formControl MuiInputBase-sizeSmall css-9ddj71-MuiInputBase-root-MuiOutlinedInput-root"
          >
            <input
              aria-invalid="false"
              class="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputSizeSmall css-1n4twyu-MuiInputBase-input-MuiOutlinedInput-input"
              id="keywordsInput"
              type="text"
              value=""
            />
            <fieldset
              aria-hidden="true"
              class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
            >
              <legend
                class="css-ihdtdm"
              >
                <span
                  class="notranslate"
                >
                  ​
                </span>
              </legend>
            </fieldset>
          </div>
        </div>
      </div>
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-8 MuiGrid-grid-sm-8 css-vok43g-MuiGrid-root"
      />
    </div>
    <div
      class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-12 css-13i4rnv-MuiGrid-root"
      style="margin-top: 1rem;"
    >
      <span
        aria-label="Search products based on provided identifiers/keywords from the Amazon Catalog."
        class=""
        data-mui-internal-clone-element="true"
      >
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium css-sghohy-MuiButtonBase-root-MuiButton-root"
          data-testid="submitButton"
          id="submitButton"
          tabindex="0"
          type="button"
        >
          Search Products
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
      </span>
    </div>
  </form>
</DocumentFragment>
`;

exports[`Test for the ProductSearchFormComponent Snapshot Test for product search form with both identifier and keyword input invalid page 1`] = `
<DocumentFragment>
  <form
    data-testid="productSearchRequestSubmit"
  >
    <div
      class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-3 css-18ksmt-MuiGrid-root"
    >
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-2 css-63b4os-MuiGrid-root"
      >
        <label
          aria-label="Comma-delimited list of product identifiers to search the Amazon catalog for"
          class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-animated MuiFormLabel-colorPrimary MuiInputLabel-root MuiInputLabel-animated css-9npbnl-MuiFormLabel-root-MuiInputLabel-root"
          data-mui-internal-clone-element="true"
          for="identifiersInput"
        >
          Identifiers
        </label>
      </div>
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6 MuiGrid-grid-sm-4 css-1xd74lj-MuiGrid-root"
      >
        <div
          class="MuiFormControl-root MuiFormControl-marginNormal MuiTextField-root css-1rv3ei8-MuiFormControl-root-MuiTextField-root"
        >
          <div
            class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-formControl MuiInputBase-sizeSmall css-9ddj71-MuiInputBase-root-MuiOutlinedInput-root"
          >
            <input
              aria-invalid="false"
              class="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputSizeSmall css-1n4twyu-MuiInputBase-input-MuiOutlinedInput-input"
              id="identifiersInput"
              type="text"
              value="UPC_testtest"
            />
            <fieldset
              aria-hidden="true"
              class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
            >
              <legend
                class="css-ihdtdm"
              >
                <span
                  class="notranslate"
                >
                  ​
                </span>
              </legend>
            </fieldset>
          </div>
        </div>
      </div>
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6 MuiGrid-grid-sm-6 css-1osj8n2-MuiGrid-root"
      >
        <div
          class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-colorPrimary MuiInputBase-sizeSmall  css-74e2me-MuiInputBase-root-MuiInput-root-MuiSelect-root"
        >
          <div
            aria-expanded="false"
            aria-haspopup="listbox"
            aria-labelledby="identifiersType"
            class="MuiSelect-select MuiSelect-standard MuiInputBase-input MuiInput-input MuiInputBase-inputSizeSmall css-jd1zyo-MuiSelect-select-MuiInputBase-input-MuiInput-input"
            id="identifiersType"
            role="button"
            tabindex="0"
          >
            UPC
          </div>
          <input
            aria-hidden="true"
            aria-invalid="false"
            class="MuiSelect-nativeInput css-yf8vq0-MuiSelect-nativeInput"
            tabindex="-1"
            value="UPC"
          />
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiSelect-icon MuiSelect-iconStandard css-pqjvzy-MuiSvgIcon-root-MuiSelect-icon"
            data-testid="ArrowDropDownIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M7 10l5 5 5-5z"
            />
          </svg>
        </div>
      </div>
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-2 css-63b4os-MuiGrid-root"
      >
        <label
          aria-label="Comma-delimited list of words or product identifiers to search the Amazon catalog for."
          class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-animated MuiFormLabel-colorPrimary MuiInputLabel-root MuiInputLabel-animated css-9npbnl-MuiFormLabel-root-MuiInputLabel-root"
          data-mui-internal-clone-element="true"
          for="keywordsInput"
        >
          Keywords
        </label>
      </div>
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-6 css-1b3l6lk-MuiGrid-root"
      >
        <div
          class="MuiFormControl-root MuiFormControl-marginNormal MuiTextField-root css-1rv3ei8-MuiFormControl-root-MuiTextField-root"
        >
          <div
            class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-formControl MuiInputBase-sizeSmall css-9ddj71-MuiInputBase-root-MuiOutlinedInput-root"
          >
            <input
              aria-invalid="false"
              class="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputSizeSmall css-1n4twyu-MuiInputBase-input-MuiOutlinedInput-input"
              id="keywordsInput"
              type="text"
              value="keywords_testtest"
            />
            <fieldset
              aria-hidden="true"
              class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
            >
              <legend
                class="css-ihdtdm"
              >
                <span
                  class="notranslate"
                >
                  ​
                </span>
              </legend>
            </fieldset>
          </div>
        </div>
      </div>
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-8 MuiGrid-grid-sm-8 css-vok43g-MuiGrid-root"
      >
        <div
          class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation0 MuiAlert-root MuiAlert-standardError MuiAlert-standard css-1hdyuqk-MuiPaper-root-MuiAlert-root"
          role="alert"
        >
          <div
            class="MuiAlert-icon css-1ytlwq5-MuiAlert-icon"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit css-1vooibu-MuiSvgIcon-root"
              data-testid="ErrorOutlineIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M11 15h2v2h-2zm0-8h2v6h-2zm.99-5C6.47 2 2 6.48 2 12s4.47 10 9.99 10C17.52 22 22 17.52 22 12S17.52 2 11.99 2zM12 20c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
              />
            </svg>
          </div>
          <div
            class="MuiAlert-message css-1pxa9xg-MuiAlert-message"
          >
            Error: Please provide either identifiers or keywords but not both.
          </div>
        </div>
      </div>
    </div>
    <div
      class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-12 css-13i4rnv-MuiGrid-root"
      style="margin-top: 1rem;"
    >
      <span
        aria-label="Search products based on provided identifiers/keywords from the Amazon Catalog."
        class=""
        data-mui-internal-clone-element="true"
      >
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium Mui-disabled MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium css-sghohy-MuiButtonBase-root-MuiButton-root"
          data-testid="submitButton"
          disabled=""
          id="submitButton"
          tabindex="-1"
          type="button"
        >
          Search Products
        </button>
      </span>
    </div>
  </form>
</DocumentFragment>
`;

exports[`Test for the ProductSearchFormComponent Snapshot Test for product search form with empty input 1`] = `
<DocumentFragment>
  <form
    data-testid="productSearchRequestSubmit"
  >
    <div
      class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-3 css-18ksmt-MuiGrid-root"
    >
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-2 css-63b4os-MuiGrid-root"
      >
        <label
          aria-label="Comma-delimited list of product identifiers to search the Amazon catalog for"
          class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-animated MuiFormLabel-colorPrimary MuiInputLabel-root MuiInputLabel-animated css-9npbnl-MuiFormLabel-root-MuiInputLabel-root"
          data-mui-internal-clone-element="true"
          for="identifiersInput"
        >
          Identifiers
        </label>
      </div>
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6 MuiGrid-grid-sm-4 css-1xd74lj-MuiGrid-root"
      >
        <div
          class="MuiFormControl-root MuiFormControl-marginNormal MuiTextField-root css-1rv3ei8-MuiFormControl-root-MuiTextField-root"
        >
          <div
            class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-formControl MuiInputBase-sizeSmall css-9ddj71-MuiInputBase-root-MuiOutlinedInput-root"
          >
            <input
              aria-invalid="false"
              class="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputSizeSmall css-1n4twyu-MuiInputBase-input-MuiOutlinedInput-input"
              id="identifiersInput"
              type="text"
              value=""
            />
            <fieldset
              aria-hidden="true"
              class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
            >
              <legend
                class="css-ihdtdm"
              >
                <span
                  class="notranslate"
                >
                  ​
                </span>
              </legend>
            </fieldset>
          </div>
        </div>
      </div>
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6 MuiGrid-grid-sm-6 css-1osj8n2-MuiGrid-root"
      >
        <div
          class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-colorPrimary MuiInputBase-sizeSmall  css-74e2me-MuiInputBase-root-MuiInput-root-MuiSelect-root"
        >
          <div
            aria-expanded="false"
            aria-haspopup="listbox"
            aria-labelledby="identifiersType"
            class="MuiSelect-select MuiSelect-standard MuiInputBase-input MuiInput-input MuiInputBase-inputSizeSmall css-jd1zyo-MuiSelect-select-MuiInputBase-input-MuiInput-input"
            id="identifiersType"
            role="button"
            tabindex="0"
          >
            UPC
          </div>
          <input
            aria-hidden="true"
            aria-invalid="false"
            class="MuiSelect-nativeInput css-yf8vq0-MuiSelect-nativeInput"
            tabindex="-1"
            value="UPC"
          />
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiSelect-icon MuiSelect-iconStandard css-pqjvzy-MuiSvgIcon-root-MuiSelect-icon"
            data-testid="ArrowDropDownIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M7 10l5 5 5-5z"
            />
          </svg>
        </div>
      </div>
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-2 css-63b4os-MuiGrid-root"
      >
        <label
          aria-label="Comma-delimited list of words or product identifiers to search the Amazon catalog for."
          class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-animated MuiFormLabel-colorPrimary MuiInputLabel-root MuiInputLabel-animated css-9npbnl-MuiFormLabel-root-MuiInputLabel-root"
          data-mui-internal-clone-element="true"
          for="keywordsInput"
        >
          Keywords
        </label>
      </div>
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-6 css-1b3l6lk-MuiGrid-root"
      >
        <div
          class="MuiFormControl-root MuiFormControl-marginNormal MuiTextField-root css-1rv3ei8-MuiFormControl-root-MuiTextField-root"
        >
          <div
            class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-formControl MuiInputBase-sizeSmall css-9ddj71-MuiInputBase-root-MuiOutlinedInput-root"
          >
            <input
              aria-invalid="false"
              class="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputSizeSmall css-1n4twyu-MuiInputBase-input-MuiOutlinedInput-input"
              id="keywordsInput"
              type="text"
              value=""
            />
            <fieldset
              aria-hidden="true"
              class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
            >
              <legend
                class="css-ihdtdm"
              >
                <span
                  class="notranslate"
                >
                  ​
                </span>
              </legend>
            </fieldset>
          </div>
        </div>
      </div>
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-8 MuiGrid-grid-sm-8 css-vok43g-MuiGrid-root"
      />
    </div>
    <div
      class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-12 css-13i4rnv-MuiGrid-root"
      style="margin-top: 1rem;"
    >
      <span
        aria-label="Search products based on provided identifiers/keywords from the Amazon Catalog."
        class=""
        data-mui-internal-clone-element="true"
      >
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium Mui-disabled MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium css-sghohy-MuiButtonBase-root-MuiButton-root"
          data-testid="submitButton"
          disabled=""
          id="submitButton"
          tabindex="-1"
          type="button"
        >
          Search Products
        </button>
      </span>
    </div>
  </form>
</DocumentFragment>
`;

exports[`Test for the ProductSearchFormComponent Snapshot Test for product search form with valid input 1`] = `
<DocumentFragment>
  <form
    data-testid="productSearchRequestSubmit"
  >
    <div
      class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-3 css-18ksmt-MuiGrid-root"
    >
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-2 css-63b4os-MuiGrid-root"
      >
        <label
          aria-label="Comma-delimited list of product identifiers to search the Amazon catalog for"
          class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-animated MuiFormLabel-colorPrimary MuiInputLabel-root MuiInputLabel-animated css-9npbnl-MuiFormLabel-root-MuiInputLabel-root"
          data-mui-internal-clone-element="true"
          for="identifiersInput"
        >
          Identifiers
        </label>
      </div>
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6 MuiGrid-grid-sm-4 css-1xd74lj-MuiGrid-root"
      >
        <div
          class="MuiFormControl-root MuiFormControl-marginNormal MuiTextField-root css-1rv3ei8-MuiFormControl-root-MuiTextField-root"
        >
          <div
            class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-formControl MuiInputBase-sizeSmall css-9ddj71-MuiInputBase-root-MuiOutlinedInput-root"
          >
            <input
              aria-invalid="false"
              class="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputSizeSmall css-1n4twyu-MuiInputBase-input-MuiOutlinedInput-input"
              id="identifiersInput"
              type="text"
              value="UPC_testtest"
            />
            <fieldset
              aria-hidden="true"
              class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
            >
              <legend
                class="css-ihdtdm"
              >
                <span
                  class="notranslate"
                >
                  ​
                </span>
              </legend>
            </fieldset>
          </div>
        </div>
      </div>
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6 MuiGrid-grid-sm-6 css-1osj8n2-MuiGrid-root"
      >
        <div
          class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-colorPrimary MuiInputBase-sizeSmall  css-74e2me-MuiInputBase-root-MuiInput-root-MuiSelect-root"
        >
          <div
            aria-expanded="false"
            aria-haspopup="listbox"
            aria-labelledby="identifiersType"
            class="MuiSelect-select MuiSelect-standard MuiInputBase-input MuiInput-input MuiInputBase-inputSizeSmall css-jd1zyo-MuiSelect-select-MuiInputBase-input-MuiInput-input"
            id="identifiersType"
            role="button"
            tabindex="0"
          >
            UPC
          </div>
          <input
            aria-hidden="true"
            aria-invalid="false"
            class="MuiSelect-nativeInput css-yf8vq0-MuiSelect-nativeInput"
            tabindex="-1"
            value="UPC"
          />
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiSelect-icon MuiSelect-iconStandard css-pqjvzy-MuiSvgIcon-root-MuiSelect-icon"
            data-testid="ArrowDropDownIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M7 10l5 5 5-5z"
            />
          </svg>
        </div>
      </div>
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-2 css-63b4os-MuiGrid-root"
      >
        <label
          aria-label="Comma-delimited list of words or product identifiers to search the Amazon catalog for."
          class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-animated MuiFormLabel-colorPrimary MuiInputLabel-root MuiInputLabel-animated css-9npbnl-MuiFormLabel-root-MuiInputLabel-root"
          data-mui-internal-clone-element="true"
          for="keywordsInput"
        >
          Keywords
        </label>
      </div>
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-6 css-1b3l6lk-MuiGrid-root"
      >
        <div
          class="MuiFormControl-root MuiFormControl-marginNormal MuiTextField-root css-1rv3ei8-MuiFormControl-root-MuiTextField-root"
        >
          <div
            class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-formControl MuiInputBase-sizeSmall css-9ddj71-MuiInputBase-root-MuiOutlinedInput-root"
          >
            <input
              aria-invalid="false"
              class="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputSizeSmall css-1n4twyu-MuiInputBase-input-MuiOutlinedInput-input"
              id="keywordsInput"
              type="text"
              value=""
            />
            <fieldset
              aria-hidden="true"
              class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
            >
              <legend
                class="css-ihdtdm"
              >
                <span
                  class="notranslate"
                >
                  ​
                </span>
              </legend>
            </fieldset>
          </div>
        </div>
      </div>
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-8 MuiGrid-grid-sm-8 css-vok43g-MuiGrid-root"
      />
    </div>
    <div
      class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-12 css-13i4rnv-MuiGrid-root"
      style="margin-top: 1rem;"
    >
      <span
        aria-label="Search products based on provided identifiers/keywords from the Amazon Catalog."
        class=""
        data-mui-internal-clone-element="true"
      >
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium css-sghohy-MuiButtonBase-root-MuiButton-root"
          data-testid="submitButton"
          id="submitButton"
          tabindex="0"
          type="button"
        >
          Search Products
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
      </span>
    </div>
  </form>
</DocumentFragment>
`;
